test/test_table.cpp

```diff
@@ -7833,5 +7833,76 @@ TEST_TYPES(Table_ColumnSizeFromRef, std::true_type, std::false_type)
     check_column_sizes(10 * REALM_MAX_BPNODE_SIZE);
 }
 
+TEST(Table_ListOfPrimitives)
+{
+    Group g;
+    TableRef t = g.add_table("table");
+    auto int_col = t->add_column_list(type_Int, "integers");
+    auto bool_col = t->add_column_list(type_Bool, "booleans");
+    auto string_col = t->add_column_list(type_String, "strings");
+    auto double_col = t->add_column_list(type_Double, "doubles");
+    auto timestamp_col = t->add_column_list(type_Timestamp, "timestamps");
+    t->add_empty_row();
+
+    std::vector<int_fast64_t> integer_list = {1, 2, 3, 4};
+    t->set_list(int_col, 0, integer_list);
+
+    std::vector<bool> bool_list = {false, false, true, false, true};
+    t->set_list(bool_col, 0, bool_list);
+
+    std::vector<StringData> string_list = {"monday", "tuesday", "thursday", "friday", "saturday", "sunday"};
+    t->set_list(string_col, 0, string_list);
+
+    std::vector<double> double_list = {898742.09382, 3.14159265358979, 2.71828182845904};
+    t->set_list(double_col, 0, double_list);
+
+    auto now = std::chrono::system_clock::now();
+    std::chrono::minutes one_minute(1);
+    std::vector<Timestamp> timestamp_list = {now, now + one_minute};
+    t->set_list(timestamp_col, 0, timestamp_list);
+
+    TableView res = t->get_list(int_col, 0);
+    auto vec = t->get_list<int_fast64_t>(int_col, 0);
```

Comment:
I know it's a unit test where it's not very important, but to help readability, it should be declared std::vector<int64_t> and not `auto`